src/MonitorProfiler/ClassFactory.cpp

```diff
@@ -16,25 +18,20 @@ STDMETHODIMP ClassFactory::CreateInstance(
     REFIID riid,
     void **ppvObject)
 {
-    if (nullptr == ppvObject)
-    {
-        return E_POINTER;
-    }
+    IfNullPointer(ppvObject);
```

Comment:
I figured out what they do, but I think the naming for these macros (`IfNullPointer` and `IfNullRet`) seemed unintuitive. That may just be due to my lack of context, but I also thought it was worth pointing out that even though they both return an error, only one has `Ret` in the name from a consistency standpoint, and I also don't know what's being returned in either case.

Reply:
`IfNullRet` is a fairly standard Windows macro (it's in corhlpr.h) that means "if the value is null, return `E_OUTOFMEMORY`". So it's good for checking if something could not be allocated and reporting such an occurance.

`IfNullPointer` is implemented in this PR to mimic `IfNullRet` except that it returns E_POINTER when the value is null. This is typically used to validate double pointers are not null (if they were null, you couldn't assign a value to the outer point becuase `*ptr` would AV). I can rename this one to something like `ExpectedPtr` which seems to be commonly used elsewhere for this.